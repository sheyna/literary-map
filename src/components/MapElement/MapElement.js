import mapOfEngland from './map-of-england.svg';
import MapLocationPoint from '../MapLocationPoint/MapLocationPoint';
import styles from "./MapElement.module.css";
import PropTypes from 'prop-types';

function MapElement(props) {
	const {locations} = props;
	return (
		<div className={styles.MapElement}>
			<img src={mapOfEngland} alt="Map of England"/>
			{locations.map((location) => {
				return (
					<MapLocationPoint 
						location={location} 
						key={location.locationId} 
						openInfoBox={props.openInfoBox}
					/>
				);
			})}
		</div>
	);
}

MapElement.propTypes = {
	locations: PropTypes.arrayOf(PropTypes.shape),
	openInfoBox: PropTypes.func,
};

export default MapElement;
