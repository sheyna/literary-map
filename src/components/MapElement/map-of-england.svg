<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 4254.09 4819.4"><defs><style>.cls-1{fill:#f2e9df;}</style></defs><path class="cls-1" d="M952.37,0s30.93,39.77,0,57.44S975.44,137,966.11,163.49s-23.41,57.44-23.41,57.44L917,141.4l-30.93,79.53-70.7,30.93s17.67,35.35,0,61.86-48.61,53-48.61,53,88.38,26.51,22.1,48.6-88.37,83.95-88.37,83.95S727,512.56,727,565.58s-4.42,92.79-26.51,92.79-22.1-66.28-22.1-66.28,0-61.86-26.51-61.86-57.44,26.52-57.44,106S660.74,782.1,687.25,782.1s53,53,53,53l61.86,22.09s-37.23-16.34-42.3-46.72-7.59-65.81-27.84-70.87-27.85-30.38-27.85-30.38,77.45,5.06,76.7,5.06,14.43-30.38,52.4,2.53,78.48,86.07,78.48,86.07l55.49,30.38s-23-12.66,0-45.57A307.88,307.88,0,0,1,1018,732s-55.69,2.53-65.82-5.06,83.54-73.41,83.54-73.41l27.84,48.09v58.23l75.94,5.06,43,65.82,22.78-27.85L1218,838.34l60.75-20.25,22.78-25.32,43-22.78L1461,800.37l25.31-38,111.39-15.19,22.78-32.91,68.35,2.53,32.9-5.06L1782.52,732l63.28-10.12-73.41,55.69L1737,744.68,1699,770l38,30.38-25.31,32.91-27.16-47.92-15.81,9.88s14.82,1,4.94,15.81-44.46,29.64-47.43,38.54,6.92,43.48-11.86,66.2S1558,964.21,1558,985s12,46.81-10,74.87-16,82.18-16,82.18l28.06,20,17.87-9-11.33,20.61L1581,1184l-6.19,15.45h15.46l3.09,21.64,20.61,21.64-13.4,12.36s13.4,33,10.3,40.18,13.4,4.12,12.37,21.64,5.15,41.21,4.12,45.33,3.09,16.49,3.09,16.49,21.64-41.21,25.76-51.52,10.3,35,9.27,44.31-13.39,35-12.36,41.21,20.6,41.21,35,42.24,26.79,8.24,36.06-15.45,19.58-45.34,31.94-56.67,6.18-22.67,6.18-22.67l10.3-19.57,27.82,10.3-1,29.88h46.41l-8.29-35,31.94,1v27.82l21.64-7.21,6.18,12.36,17.52,7.21-7.21,12.37,7.21,19.57-35,37.89-36.06-4.92-30.91,60.79h48.42l26.79-26.79-10.3,26.79-38.12,24.73,37.09,12.36-50.49,7.21-22.67-12.36-14.42,3.09-6.18,30.91-16.49-38.12L1755.09,1558l9.27,25.76-17.51,17.52,9.27,20.6-15.46,27.82,37.1,27.82,9.27,43.27,16.48-14.42,16.49-3.09,12.36,11.33,55.64-27.82,36.06,6.19-48.42,20.6-16.49,7.21-7.21,13.4-55.64,7.21-31.94,72.12-22.67,7.22-9.27,19.57,18.55,15.46-20.61,10.3,47.4,87.58,15.45,21.63,36.06,11.34-3.09,17.51,7.21,15.46,29.88-6.18,5.15-18.55,22.67,3.09,2.06,16.49-17.51,14.42,9.27,16.48-28.85,10.31-29.88-17.52-15.45,3.09,3.09,14.43-44.31-50.49-3.09-29.88-8.24-23.69-13.39-33-27.82,1-6.18,14.42-36.07,16.49-20.6,26.78,36.06,46.37,7.21,31.94,15.46,12.36,58.72,55.64,43.28,34-67-21.64-60.79-39.15-27.82-4.12-14.43-33-21.63-8.24,2.06-18.55-30.91-27.82-40.18,8.25L1514,2000l-26.78,9.27-13.4,41.22-58.73-39.16-60.78-23.69,14.42,39.15-50.49,28.85-18.54-3.09-22.67,20.6-38.12,19.58-6.18-17.52s-52.55,26.79-76.25,34-44.3,23.7-44.3,23.7l-12.36,15.46-23.7-1-1,40.18,14.42,28.85-24.72-9.28-20.61,15.46-1,32.6-14.43,4.49s5.16,24.73-11.33,31.94-44.3,0-44.3,0-11.34,29.88-20.61,33-52.55,12.37-52.55,12.37,4.13,22.66,4.13,36.06-39.16,47.39-39.16,47.39L901,2456.41,944.23,2409l11.33,9.27,28.85-12.36,17.52,15.45,18.54,2.06,4.12-16.48L1011.2,2375l34-20.61,1-17.51,65.94-21.64,47.39-1,53.58,13.39,8.24,16.49-19.57,6.18s13.39,20.61,4.12,33-24.73,20.61-24.73,20.61,22.67,15.45,23.7,27.81-6.18,18.55-6.18,18.55l31.94,23.7s-1,29.88-15.46,50.48-45.33,46.37-45.33,46.37,39.15,76.24,29.88,98.91-9.28,55.64-9.28,55.64l-16.48,4.12-15.46,28.85,11.34,18.54-27.82,24.73v26.79L1105,2851l-15.46,30.91-41.21,17.52,1,11.33h-41.21l-55.64,46.37-12.36-13.4-19.58,13.4-20.61-12.37-26.78,4.12,10.3,18.55-21.64,8.24,15.46,28.85-29.88,7.21,10.3,38.13-17.51,13.39-25.76-31.94-22.67,26.79,1,13.39-31.94-5.15,1-20.61-26.79-1,8.25,22.67-15.46,29.88h-33l-84.49,46.36v19.58l50.49-11.33,5.15,16.48s17.52-5.15,27.82,0,19.58,29.88,15.46,40.18-5.16,37.1-5.16,37.1l-55.63,36.06S664,3287.88,665,3292s2.06,27.82,10.3,25.76l8.25-2.06-3.09-12.37v-18.54l27.82,13.39,17.51-10.3,47.4,19.58,18.54,3.09,19.58-7.22-16.49,24.73H755.68l-12.36-10.3L733,3328.06l-7.21-15.45-17.51-1-14.43,46.36s28.85,1,29.88,14.43-1,43.27-1,43.27,50.49,8.24,61.82,8.24,9.27-29.88,35-39.15,67,24.73,67,24.73l12.36-28.85-10.3-58.73,28.85-13.39,8.24,14.42,90.67,1,8.24,18.55,36.06-6.19,14.43,7.22-33,15.45s24.73,1,24.73,13.4,8.24,20.6,8.24,20.6,14.42-5.15,27.82-4.12,34,8.24,34,8.24,2.06,18.55-16.49,26.79-39.15,12.37-39.15,12.37l3.09-19.58s-24.73,15.45-28.85,19.58S1037,3463,1037,3463s10.3-18.54,23.69,0,21.64,13.4,42.25,13.4,30.91.65,30.91.65l-9.28-20.23,22.67-2.06,17.52,22.29s0-22.29,12.36-21.26-4.12-2.06,22.67-25.76,87.57-38.12,87.57-38.12,8.25,37.09,39.16,74.18,61.82,59.76,54.6,76.25,18.55,52.54,18.55,52.54,40.18-2.06,52.55,0,37.09,19.58,37.09,19.58l63.88-5.15,31.94,16.48,27.82-21.63-14.43-24.73,16.49-23.7,1-23.69,12.36,11.33,38.12-21.64,11.34,14.43,34-45.34,9.27,22.67,28.85,12.36-8.25-24.72s15.46,10.3,24.73,2.06,73.16-46.37,73.16-46.37l12.36,8.24,15.45-26.78,25.76-2.06,30.91-39.16,11.33-5.15,6.18-22.67,52.55-25.75s25.76,3.09,28.85-23.7c5.15,25.76-5.15,35-26.79,43.27s-42.24,13.4-48.42,33-1,41.22-21.64,51.52-57.7,25.76-64.91,46.36-23.7,51.52-23.7,51.52l11.34,25.76-16.49-2.06-8.24-17.52h-21.64l-10.3-6.18s-12.36,55.64-15.45,59.76-23.7,50.48-23.7,50.48-18.55-1-34,16.49-18.55,26.79-18.55,26.79l24.73,36.06-30.91-13.4s-2.06,33-2.06,41.22,18.55,45.33,18.55,45.33l-18.55,4.12-10.3-23.7-11.34,28.85-56.66-5.15-5.16-14.42-18.54,19.57-15.46-19.57-16.48,19.57-53.58,13.4h-28.85s-8.24-14.43-35-26.79-45.33-7.21-60.79-6.18-38.12-18.55-64.91-18.55-68,9.27-78.3,9.27-25.76,1-51.52,2.07-50.48-5.16-57.7-5.16-25.75-3.09-31.94-3.09-8.24,15.46-8.24,22.67v20.61l-23.69-16.49-19.58-3.09s19.58,0,19.58,24.73,6.18,41.21,6.18,41.21h-15.46s16.49,4.12-18.54,28.85-58.73,27.82-80.37,19.58a103.12,103.12,0,0,0-42.24-6.19l5.15,19.58-11.33,17.51-27.82,5.16,16.48,23.69-15.45,19.58s6.18,5.15,5.15,22.67-15.45,38.12-15.45,38.12l5.15,30.91L867,4109s-52.54,16.49-58.72,16.49-30.91,7.21-36.07,7.21-46.36,13.4-49.45,28.85-11.33,38.12-27.82,38.12-27.82-13.39-27.82-13.39-33,15.45-52.54,45.33-40.19,59.76-40.19,59.76l-75.21,21.64v19.57H482.65l-8.25,20.61H448.64L411.55,4379l-9.27-14.43L357,4359.41l-6.19,24.73-17.51,4.12v16.49H311.61l-9.27-21.64s-19.58,1-74.18,5.15-81.4,16.49-81.4,16.49,7.21,35,5.15,42.24-5.15,27.82-5.15,27.82l58.73,21.63,49.45-54.6,24.73,4.12s7.21,25.76,9.28,28.85,57.69,18.54,53.57,45.33,1,69,1,80.37S358,4641.72,358,4641.72l74.18-19.58s-16.49-65.94-9.27-75.21,24.72,1,25.75-17.52,23.7-16.48,12.37-35c15.45-13.39,15.45-27.82,29.88-27.82s37.09-3.09,37.09-3.09,5.15,23.7-14.43,22.67-29.87,6.18-30.9,13.39,13.39,8.25,9.27,27.82,16.48,23.7,21.63,6.18,19.58-45.33,44.31-50.48,33-1,39.15,3.09,21.64,21.64,21.64,21.64l18.54-19.58s-9.27-11.33,2.06-37.09,30.91-16.49,39.16-21.64,13.39-22.66,13.39-22.66l22.67-2.06s2.06,18.54-3.09,29.87,3.09,19.58,3.09,19.58,12.36-3.09,39.15-2.06,59.76,6.18,59.76,6.18-9.28-21.63,17.51-36.06,40.19-19.58,40.19-19.58l21.63,24.73a57.73,57.73,0,0,1,34,8.24c16.49,10.31,24.73,53.58,24.73,53.58l15.46-6.18,18.54-30.91,29.88,3.09,44.3,64.91,65.94,24.73,9.28,14.42-11.34,14.43s48.43,33,68,34,46.37,12.36,46.37,12.36-8.24-57.7,6.18-67,50.49,28.85,58.73,17.52,42.24-49.46,42.24-49.46,7.21-10.3-13.39-13.39-20.61-30.91,3.09-47.4,21.64-25.76,24.73-47.39,48.42-37.09,54.6-54.61,7.22-60.79,7.22-60.79l17.51,30.91,36.06-2.06s14.43-9.27,30.91-27.82,44.31-28.85,63.88-31.94,88.61-44.3,88.61-44.3l14.42,9.27s11.34-11.33,26.79-11.33,51.52-17.52,64.91,2.06,16.49,35,55.64,46.36,75.21,55.64,87.58,65.94c5.22,4.35,22.84,11.88,31.6,22.52,12,14.56,9.85,31,19.42,33,1.72.25,17.8-9.24,18.26-13.91.33-3.38,0-17.11-12.18-18.55s-18-2.27-31.3-11c-22.23-14.66-49.77-37-49.85-38.55-.09-1.87,30.78,20.81,53,32.17,10.33,5.27,16.29-1.77,16.23-2.61-.72-10.53,4.23-12.37,4.23-12.37l-7.21-21.64,19.58-14.42L2088.91,4310l14.43,15.45h19.57l57.7,22.67v17.51l41.21,15.46,12.37-17.52-10.31-16.48,10.31-25.76-10.31-5.15,1-22.67-29.37,15.46-29.35-40.19,10.92-21.84-5.24-9.36s-20.54,27.67-28.55,14.41,16.84-35.33,16.84-35.33,16-19.87,21.26-17.39,1.87,35.31,1.87,35.31l4.53-9.07,46.37,21.64,6.18,31.94,29.88-15.46,15.45,11.33,40.19-8.24L2310.43,4244l21.64,6.18,52.54-6.18,21.64,29.87-5.15-30.9,20.61-18.55,28.84-34,49.46-4.12,56.67,5.15-23.7-18.55-6.18-20.6-46.37-38.12v-15.46l-5.15-27.82,24.73,24.73,24.73,20.61,24.72,41.21,50.49-4.12,24.73,22.66,29.88-3.09,38.12-38.12,48.42,6.18,24.73,26.79,27.82-8.24-11.33,22.67L2737,4193.53s25.76,18.55,46.37,23.7,37.09,23.69,37.09,23.69l15.45-33s51.52-8.24,65.94-13.39,43.72-3.48,43.72-24.15,25.84,20.67,25.84,20.67l50-13.78,8.61-37.9,55.13-8.61s70.62,3.45,99.91,27.56,41.34,58.57,99.91,58.57S3371,4229,3371,4229l10.33-43.07s70.63-20.67,113.7-39.62,98.19-58.57,139.53-74.07,65.46,1.72,65.46,1.72L3738,4055s-31-51.68,0-87.86,86.13-5.17,141.25-75.79,48.23-175.71,48.23-175.71l-43.06-12.06,53.4-32.73v-41.34s-12.06,22.39-55.12,22.39-118.86-10.33-118.86-10.33l-82.69,53.4-29.28-3.45,31-29.28-17.22-18.95-63.74-17.23-8.61-22.39-34.46-3.44-29.28,12.05-62-13.78s8.61-24.11,29.28-24.11,43.07-10.34,43.07-10.34S3564,3552,3576,3560.61s36.18,5.16,36.18,5.16,27.56-8.61,27.56-15.5,31.77-22.65,31.77-37.43-.74-44.33-.74-44.33l-20.69-1.48s-3,6.65-13.3,6.65-47.29-8.13-47.29-8.13,48-5.91,51.72-7.39,23.65-7.39,23.65-7.39-1.48-22.9-1.48-42.85-2.21-48.77-2.21-48.77-14,12.56-45.08,12.56-68-8.87-68-8.87,36.21-5.91,51-17,32.51-17,36.94-17,36.21,8.13,38.43-22.17c16.25.74,14.77,21.43,25.86,22.91s35.47,7.39,35.47,7.39,40.64-11.82,53.94-42.12,24.38-37,24.38-56.9-26.6-14.78-16.25-29.55-.74-23.65-.74-27.34,15.52-13.3,14.78-23.65-6.65-19.21-6.65-19.21-7.39,13.3-23.65,14-36.21,5.17-39.9,5.17,23.64-8.13,35.47-17.73,20.69-18.48,20.69-18.48l-25.13-26.6s44.34,20.69,46.56,25.13,14,27.34,14,27.34,6.65,11.08,9.6-6.65-5.91-46.19,3-60.23,14-45.07,19.21-56.16,32.51-10.34,32.51-10.34,21.43-68,31-76.85-14.78-30.29-2.22-51.72,21.43-33.25,29.56-48.77,36.2-37,30.29-49.51-25.86-.74-25.86-11.08,5.91-38.43,3.69-51-11.08-11.08-2.21-41.38-6.65-68-11.09-82.76-26.6-28.08-33.25-53.2-28.82-51.73-48-51.73-34-41.38-53.2-40.64c-21.19,1.94-18.1-13-27.72-19.37-6.65-4.37-12.83-1-17.53-5.5-5.7-5.46-3.74-8.74-7.9-11.68-6.07-4.28-7.07,3.75-16.15-3.09-9.72-7.33-12.84-15-20.1-14.22-14,1.48-15.93,8.19-31.45-8.81s-13.06-1.72-66.84-11.22c-12.95-2.69-14.42,2.12-24.92,1.94-21-5.15-48.11-6.87-67.44-6.37-7.66-1.21-17.08-9.33-29.13-11.84-2.47-.43-4.35,6.24-13.74,5.84-11.69-6.19-29-8-36.77-8.93-5.13-.63-5.15,5.9-9.63,5.15-8-1.34-9-13.07-14.92-12.39-25.86,3-96.8,12.56-96.8,16.26s-1.19,18.65,2.1,30.14,3.28,35.31.82,44.34S3335,2461.75,3335,2479s4.93,28.74,4.93,28.74-13.14-8.21-14,3.28a48.6,48.6,0,0,1-5.75,19.71l-19.71-23s-46-5.75-56.65,0-29.57,22.17-29.57,22.17l-5.74-21.35-73.08-4.93-9-9-29.56,4.11-11.49-34.49,25.45,3.28,10.37-16.77,14.18-8.06,4.84-24.18,11.93-5.48,5.49-16.94L3142,2374.71l31.21-.83a36.88,36.88,0,0,1,6-15.37c2.61-3.67,8.76-4.36,11.29-9,2.26-4.19-.74-5.19,1-9.68,7.41-6.44,10-17.48,13.86-22.89,3.56-4.93,6.86-5.15,9.77-7.89,14.78-14,46-53.89,46-53.89l-18.86-2.42-5.29,9.07L3225,2250.49l46.75-8-32.89-5.55s29.49-9.73,19.67-5.16,10.76-9,10.76-9a72.7,72.7,0,0,1-1.41-21.32c3.94-5.18,5.48-13.54,1.61-21.92.46-5.95,4.73-10.2,4.73-13.42s-6-8.45-9.24-14.95c-3.38-6.9-4-15.09-6.77-23.22,1.61-15.15-6.46-33.46-10.27-40.66-1.85-3.51-12.59-8.06-19.39-14.47-3.86-3.64-3-9.13-7.1-12.26-11.93,1.62-15.94-7.81-17.41-14.5-2.68-2.47-4.91,4.77-6.77,1.93-3.06-4.67,1.64-14.16,1-19.35-1.23-9.54-4.19-5.77-11.77-13.93-4.6.76-8.45-8.75-17.25-18-10.64-11.16-26.69-23.29-36.24-34.24-2.36-2.71-14.7-3.48-16.64-20.25-8.59-8.43-16.16-13.14-15.8-22.24-5.54-6.13-10.31-4.74-11.93-6.77-8.42-11.18-6.12-16.13-6.12-16.13-9.06-9.86-17.48-11-26.44-16.76-10.2-6.6-20.83-17.21-29-17.21-30.39,0-25.46-22.17-35.31-46s-22.17-47.62-22.17-47.62-17.25,13.14-27.92,13.14-23-20.53-46.81-16.43-27.92,15.61-53.37,1.65,4.1-14.78,23.81-20.53,45.16,2.46,66.51-4.11,50.09-15.6,68.16-1.64,50.09,40.24,85.4,47.63,43.52,36.13,57.48,44.34,39.41,37.77,39.41,37.77-12.32-4.93-14.78-36.13-23-73.08-40.24-96.07-34.48-36.13-34.48-36.13-20.53-61.59-32-102.64-1.64-47.63-4.92-65.7-23-36.13-7.39-44.34,9-25.45,13.13-43.52,18.89-4.1,37.78-8.21,27.91-4.1,27.91-4.1l-13.95-23.82s-20.53,4.11-26.28,4.11-23-8.21-23-8.21l-2.47-16.43-27.09-11.49s-5.75-18.07-5.75-23.81-7.39-1.65-17.25-3.29-26.27-20.53-32.84-27.1,1.64-14.78,1.64-14.78l4.93-16.42-26.28-1.64,4.93-17.25s-14-4.1-14-18.06-14-32-14-32l4.93-34.48-46.81-42.7-10.67,17.24-11.5-35.31s-14.78-7.39-30.38-6.57-20.53-23.81-48.45-18.06-38.59-9.85-46.8-10.68-30.38-21.35-44.34-22.17-48.45-11.49-48.45-11.49l-37,11.49s4.92-22.17,4.92-32-16.42-37.77-16.42-37.77l19.71-7.39-2.46-11.5-28.74-.82,1.64-19.71-12.32-9.85s4.11-34.49.82-41.06-12.31-17.24-12.31-17.24,2.46-25.46,2.46-30.38-12.32-46-12.32-46l14-.82s1.64-27.92,1.64-31.2-1.64-16.42,9-25.46-4.93-23-4.93-23L2559,730.79l9-7.39-15.6-21.35-14.78-12.31L2540.1,629l-15.6-7.39s16.42-3.28,18.89-14.78-11.5-24.63-11.5-24.63,19.71-9,28.74-15.6-5.75-17.25-5.75-17.25,6.57-14.78,6.57-22.17-21.35-18.06-25.45-18.06,2.46-23-3.29-39.42-3.28-41.88-3.28-41.88l-18.89-11.49s14-13.14,15.6-16.43-3.28-46.8,7.39-64.87-5.74-38.59-7.39-41.05S2522,257,2513,248.79s-20.52,1.64-32.84-19.71a278.46,278.46,0,0,1-18.89-39.42L2449,202s-46.81-23.81-46.81-60.76-17.24-31.21-17.24-47.63-9.86-20.53-9.86-20.53,4.11-5.75-1.64-25.45-13.14-30.38-15.6-38.6S2343,0,2343,0Z"/><path class="cls-1" d="M984.91,1963.69,948,1921.29l-17.78,15,6.84,27.36,20.51,19.14,27.36,9.58,13.68,23.25s-8.21,31.46,15,58.82,49.24,64.28,68.39,56.07,58.81-36.93,80.7-43.76,39.66-9.58,56.08-41,42.4-52,42.4-52l-31.46-26-36.93,9.57-19.15,34.2-27.35-90.27s-16.42-27.36-16.42-35.57-34.19,0-34.19,0v-20.51l-86.17-5.47-5.47,94.37Z"/><path class="cls-1" d="M939.46,1336.36c10.45-1,55.38-17.76,65.83,5.23s5.22,57.46,8.35,57.46,20.9,10.45,20.9,10.45-15.67-72.09,17.76-75.23,43.4-9.4,43.4-9.4l-20.41-29.26s33.43,7.32,97.17-32.39,143.14-63.73,143.14-63.73l-23-41.79,32.39-13.59,10.45-34.48L1229.92,1117l-15.67,26.12,8.36,12.54s-20.9-1-31.34,6.27S1161,1189,1161,1189l-35.53-2.09-93,20.9L1022,1259.57l-37.61,8.88-9.4,17.76-28.21-1-16.58,35.52Z"/><path class="cls-1" d="M2418.77,4269.29s-2.86,29.44-6.13,36.8.41,31.48,13.08,28.21,26.58-21.67,41.71-15.95,5.31,29,31.89,36,22.08,32.3,22.08,32.3,21.26,14.31,54.38,6.54,31.07-3.27,36.38-24.94,9-27.39,26.17-27,47.43,7.77,47.43,7.77l8.18-21.67s-16.77-1.64-16.77-12.27,7.77-29.85,7.77-29.85l-24.12,3.68-8.59-16.35L2640.38,4284s-29.44,12.26-40.89-4.09-2.45-45.18-2.45-45.18a76.18,76.18,0,0,0-15.95,6.54c-7,4.09-19.62,24.12-19.62,24.12l-4.5-15.94s-27.39,7.77-55.61,15.94-59.28,10.23-65.41,2.05S2418.77,4269.29,2418.77,4269.29Z"/></svg>