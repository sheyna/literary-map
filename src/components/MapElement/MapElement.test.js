import ReactTestRenderer from 'react-test-renderer';
import MapElement from  './MapElement';
import locations from '../../locations.json';

describe('MapElement component', () => {
  it('should match snapshot', () => {
    const component = ReactTestRenderer.create(
      <MapElement locations={locations}/>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});