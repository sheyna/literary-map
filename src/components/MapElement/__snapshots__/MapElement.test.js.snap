// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapElement component should match snapshot 1`] = `
<div
  className="MapElement"
>
  <img
    alt="Map of England"
    src="map-of-england.svg"
  />
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "49%",
        "top": "30%",
      }
    }
  >
    <img
      alt="BrontÃ« Country"
      src="img/brontes-thumb.png"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "27%",
        "top": "77%",
      }
    }
  >
    <img
      alt="A Romance of Exmoor"
      src="img/lorna-doone-thumb.png"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "50%",
        "top": "58%",
      }
    }
  >
    <img
      alt="The Bard of Stratford-upon-Avon"
      src="img/shakespeare-thumb.png"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "38%",
        "top": "20%",
      }
    }
  >
    <img
      alt="William Wordsworth's home"
      src="img/wordsworth-thumb.png"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "60%",
        "top": "22%",
      }
    }
  >
    <img
      alt="Count Dracula Arrives in England"
      src="img/dracula-thumb.png"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "70%",
        "top": "72%",
      }
    }
  >
    <img
      alt="London's Writers' Block"
      src="img/cheyne-thumb.png"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "52%",
        "top": "37%",
      }
    }
  >
    <img
      alt="Romance in the English Civil War"
      src="img/peveril-peak-thumb.png"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "46%",
        "top": "40%",
      }
    }
  >
    <i
      className="fas fa-thumbtack"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "48%",
        "top": "38%",
      }
    }
  >
    <i
      className="fas fa-thumbtack"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "62%",
        "top": "78%",
      }
    }
  >
    <i
      className="fas fa-thumbtack"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "49%",
        "top": "55%",
      }
    }
  >
    <i
      className="fas fa-thumbtack"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "62%",
        "top": "83%",
      }
    }
  >
    <i
      className="fas fa-thumbtack"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "45%",
        "top": "85%",
      }
    }
  >
    <i
      className="fas fa-thumbtack"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "17%",
        "top": "87%",
      }
    }
  >
    <i
      className="fas fa-thumbtack"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "53%",
        "top": "70%",
      }
    }
  >
    <i
      className="fas fa-thumbtack"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "39%",
        "top": "85%",
      }
    }
  >
    <i
      className="fas fa-thumbtack"
    />
  </div>
  <div
    className="MapLocationPoint"
    onClick={[Function]}
    style={
      Object {
        "left": "75%",
        "top": "80%",
      }
    }
  >
    <i
      className="fas fa-thumbtack"
    />
  </div>
</div>
`;
